### ISSUE TITLE
Config structure for using multiple transformers with source-filesystem

### ISSUE THREAD

**Issue by namoda378** [gridsome/gridsome#401]
## Description

The current configuration file structure seems a bit awkward when using multiple transformers with source-filesystem plugin.

### What I Expected to Work

<pre>
{
      use: '@gridsome/source-filesystem',
      options:[
        {
          path: 'markdowns/**/*.md',
          typeName: 'Markdown',
          remark: {
            plugins: [
              // ...local plugins
            ]
          }
        },
        {
          path: 'yamls/**/*.yaml',
          typeName: 'Yaml',
          yaml: {
            plugins: [
              // ...local plugins
            ]
          }
        },
      ] 
    },
</pre>

### What Actually Works

<pre>
 plugins:[ {
      use: '@gridsome/source-filesystem',
      options: 
        {
          path: 'markdowns/**/*.md',
          typeName: 'Markdown',
          remark: {
            plugins: [
              // ...local plugins
            ]
          }
        }
    },
    {
      use: '@gridsome/source-filesystem',
      options: 
        {
          path: 'yamls/**/*.yaml',
          typeName: 'Yaml',
          yaml: {
            plugins: [
              // ...local plugins
            ]
          }
        }
    }],
</pre>

I know this is a very trivial issue but this seems more consistent with how webpack configs are written. I find it a bit awkward, never mind if it's just me.

**Comment by kaceo** [comment]
Yes it seems quite awkward, and also the gridsome.config.js can get rather complicated and hard to read in a project with many data sources (e.g. an ecommerce site, dozens of schemas).
